src/itwinai/loggers.py

```diff
@@ -350,13 +374,18 @@ def create_logger_context(self, rank: Optional[int] = None):
         if not self.should_log():
             return
 
-        os.makedirs(self.savedir, exist_ok=True)
-        run_dirs = sorted([int(dir) for dir in os.listdir(self.savedir)])
-        if len(run_dirs) == 0:
-            self.run_id = 0
+        # os.makedirs(self.savedir, exist_ok=True)
+        if os.path.isdir(self.savedir):
+            numeric_dirs = [
+                int(name)
+                for name in os.listdir(self.savedir)
+                if os.path.isdir(os.path.join(self.savedir, name)) and name.isdigit()
+            ]
```

Comment:
I think this section would be a bit cleaner with pathlib, e.g. if `self.savedir` is a Path object then you can just do `(self.savedir / name).is_dir()` which is considerably shorter.